### PR TITLE
End to end encryption 2

### PR DESCRIPTION
**Pull request by bluquar**
Ensure that the user is always logged in and has active end-to-end encryption credentials if we're on the Home/Profile tab screen

https://user-images.githubusercontent.com/4309735/144341571-c748ac2c-21cd-4cc5-9784-e14c50dc4522.mov

### CODE REVIEW

**Comment by bluquar** [comment]
Added progress bar since creating the credentials can take a while, and there are several steps so we can show a meaningful progress bar.

https://user-images.githubusercontent.com/4309735/144345647-1f5f11a9-3c40-4d59-82e8-09c2286dd914.mov

**Comment by lowellbander** [comment]
Does this mean it's not possible to view requests without logging in?

**Comment by lowellbander** [comment]
I'm thinking it'd be really valuable for first timers to see the requests in the app without logging in, and would make it more likely for folks to want to create an account

**Comment by lowellbander** [comment]
I guess this isn't an encryption specific comment, more authentication generally

**Comment by bluquar** [comment]
Currently it's not possible to view requests without logging in. I think it would make sense to implement a "View Only" mode that has a CTA to create an account, which would then redirect you to the current register/login flow

**Comment by bluquar** [comment]
In fact probably the easiest way to do that would be to implement HomeScreen and ProfileScreen *not* assuming that the user is logged in, since most things would probably work the same, and then conditionally render/handle things where logged in and not-logged-in behavior are different.

**Comment by bluquar** [comment]
I would still like to merge this and then I'll make that change in a followup PR though :)

**Comment by bluquar** [comment]
Here's my proposal for auth flow: https://www.figma.com/file/CViOmNtL3UyCLAhgWd9Rtk/Authentication-Flow?node-id=0%3A1

File: src/ca/screens/PersonDetailsScreen.tsx [hunk 1]
```diff
@@ -37,14 +38,19 @@ export default function PersonDetailsScreen({
...
   React.useEffect(() => {
+    const title =
+      resolvedPersonNumber == null ? '' : `Person ${resolvedPersonNumber}`;
     navigation.setOptions({
-      title:
-        resolvedPersonNumber == null ? '' : `Person ${resolvedPersonNumber}`,
+      title,
     });
+    if (title !== '') {
+      logToConsoleWhatTheEncryptedTitleWouldBeAndDecryptIt(seald, title);
```
Comment (lowellbander) [inline src/ca/screens/PersonDetailsScreen.tsx:52]: fuckin nice, love me some descriptive function names

**Review by lowellbander — APPROVED** [review]
filed #14 for view only!

**Pull request merged**